src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java

```diff
@@ -925,39 +893,29 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
     }
 
     @SmackIntegrationTest
-    public void isEqualToOneTest() {
-        assertEquals(1,1);
-    }
-
-    @SmackIntegrationTest
-    public void sendMessageEventTest() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, XmppStringprepException, TestNotPossibleException {
-        final String nodename = "sinttest-publish-item-nodename-" + testRunId;
-        final String needle = "test content " + Math.random();
+    public void receiveOfflineMessageEventItemTest() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, ExecutionException, PubSubException
+    {
+        final String nodename = "sinttest-receive-published-item-nodename-" + testRunId;
+        // final String needle = "test content " + Math.random();
         LeafNode publisherNode = pubSubManagerOne.createNode(nodename);
         try {
-            // Publish a new item.
-            publisherNode.publish( new PayloadItem<>( GeoLocation.builder().setDescription( needle ).build() ) );
-            Node subscriberNode = pubSubManagerTwo.getNode(nodename);    
-            ChatManager chatManager = ChatManager.getInstanceFor(connection);
+            final Node subscriberNode = pubSubManagerTwo.getNode(nodename);
             final EntityBareJid subscriber = conTwo.getUser().asEntityBareJid();
             subscriberNode.subscribe( subscriber );
-            Message m = new Message();
-            // Retrieve items and assert that the item that was just published is among them.
-            MessageEvent message = new MessageEvent();
-            message.setOffline(true);
-            message.setStanzaId(StanzaIdUtil.newStanzaId());
-            m.addExtension(message);
-            // final List<Item> items = publisherNode.getItems();
-            System.out.println(message.toString());
-            System.out.println(m.toString());
-
-            // EntityBareJid jid = JidCreate.entityBareFrom("itot@goodbytes.im");
-            Chat chat = chatManager.chatWith(subscriber);
-            chat.send(m);
-            // assertTrue( items.stream().anyMatch( stanza -> stanza.toXML( "" ).toString().contains("offline") ) );
-        }
-        catch (XMPPErrorException | PubSubException.NotAPubSubNodeException xe){
-            throw new TestNotPossibleException( "CANT DO IT" );
+            Message message = new Message();
+            MessageEvent event = new MessageEvent();
+            event.setOffline(true);
+            event.setStanzaId(StanzaIdUtil.newStanzaId());
+            message.addExtension(event);
+            final CompletableFuture<Stanza> result = new CompletableFuture<>();
+            conTwo.addAsyncStanzaListener( result::complete, stanza -> stanza.toXML("").toString().contains("offline"));
+            publisherNode.publish( new PayloadItem<>( GeoLocation.builder().setDescription(message.toXML("").toString()).build() ) );
```

Comment:
What you are publishing is:

- A pubsub item
- that has a payload of type 'geolocation'
- that in the _description_ of the geolocation has the XML representation of a message, that has no addressing, but has a message event extension.

This is the stanza that is being sent to publish the data by this test:

```<iq to='pubsub.laptop-guus' id='IP82V-348' type='set'>
  <pubsub xmlns='http://jabber.org/protocol/pubsub'>
    <publish node='sinttest-receive-published-item-nodename-lm7vp'>
      <item xmlns='http://jabber.org/protocol/pubsub'>
        <geoloc xmlns='http://jabber.org/protocol/geoloc'>
          <datum>
            WGS84
          </datum>
          <description>
            &lt;message xmlns=&apos;jabber:client&apos; id=&apos;IP82V-346&apos;&gt;&lt;x xmlns=&quot;jabber:x:event&quot;&gt;&lt;offline/&gt;&lt;id&gt;IP82V-347&lt;/id&gt;&lt;/x&gt;&lt;/message&gt;
          </description>
        </geoloc>
      </item>
    </publish>
  </pubsub>
</iq>
```

This is a very, very weird construct. I don't think you intended this.

The code still misses documentation, so it is hard for me to guess what exactly it is that you want to test.

If you provide the purpose of the test (in the javadoc of the method), then I can better understand exactly what it is that you want to do.